### PR TITLE
Adds ZTP configuration documentation

### PR DESCRIPTION
**Pull request by alosadagrande**
Includes the last part of the documentation: configure ZTP for the factory-cli-tool

Signed-off-by: Alberto Losada <alosadag@redhat.com>

### CODE REVIEW

**Comment by alosadagrande** [comment]
cc/ @donpenney @browsell @obochan-rh

**Comment by obochan-rh** [comment]
When using the ZTP we need to add a Release notes regarding the limitation of the rootfs isn't loaded from the local mount  and and downloaded from the registry.

File: docs/ztp-precaching.md [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+
+Zero Touch Provisioning (ZTP) leverages multiple products or components to deploy OpenShift Container Platform clusters using a GitOps approach. While the workflow starts when the site is connected to the network and ends with the CNF workload deployed and running on the site nodes, it can be logically divided into two different stages: provisioning of the SNO and applying the desired configuration, which in our case is applying the validated RAN DU profile.
+
+> :warning: The workflow does not need any intervention, so ZTP automatically will configure the SNO once it is provisioned. However, two stages are clearly differentiated.
+
+The workflow is officially started by creating declarative configurations for the provisioning of your OpenShift clusters. This manifest is described in a custom resource called `siteConfig`. See that in a disconnected environment, there is a need for a container registry which has been configured to deliver the required OpenShift container images required for the installation. This task can be achieved by using https://docs.openshift.com/container-platform/latest/installing/disconnected_install/installing-mirroring-disconnected.html[oc-mirror].
```
Comment (donpenney) [inline docs/ztp-precaching.md:41]: The URL here ends up in a bad link with the [oc-mirror] at the end

File: docs/ztp-precaching.md [hunk 2]
```diff
@@ -0,0 +1,302 @@
...
+
+![ZTP workflow 1](images/ztp_workflow_1.png "ZTP workflow 1")
+
+The provisioning process includes installing the host operating system (RHCOS) on a blank server and deploying OpenShift Container Platform. This stage is managed mainly by a ZTP component called the Infrastructure Operator
+
+> :exclamation: Notice, in the picture, how ZTP allows us to provision clusters at scale.
+
+![ZTP workflow 2](images/ztp_workflow_2.png "ZTP workflow 2")
+
+Once the clusters are provisioned, the day-2 configuration defined in multiple `PolicyGenTemplate` (PGTs) custom resources will be automatically applied. `PolicyGenTemplate` custom resource is understood by the ZTP using a specific kustomize plugin called https://github.com/openshift-kni/cnf-features-deploy/tree/master/ztp/policygenerator-kustomize-plugin[policy-generator]. In telco RAN DU nodes, this configuration includes the installation of the common telco operators, a common configuration for RAN and specific configuration (SR-IOV or performance settings) for each site since it is very dependant on the hardware.
```
Comment (donpenney) [inline docs/ztp-precaching.md:65]: Similar bad link syntax

File: docs/ztp-precaching.md [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+
+> :warning: The number of container images extracted and loaded is way bigger than in the discovery stage. So, depending on the OCP release and whether it was requested to install the telco operators, the time it takes will vary. 
+
+* There are two systemd units (var-mnt.mount and precache-ocp.services). The precache-ocp.service depends on the disk partition to be mounted in /var/mnt by the var-mnt unit. The precache-ocp service basically calls a script called `extract-ocp.sh`. Notice that the precache-ocp must be executed before the machine-config-daemon-pull.service and nodeip-configuration.service, so it means that extracting the images is done before the OCP installation starts.
+* The `extract-ocp.sh` script basically uncompresses and loads the images required in this stage from the disk partition to the local container storage. Fundamentally there are OCP release images and operators if they were requested to be installed. Once done, images can be used locally instead of pulled down from a registry.
```
Comment (donpenney) [inline docs/ztp-precaching.md:258]: Are the decoded extraction scripts stored somewhere? Maybe they can be added into the repo with a reference here, noting that they're encoded in this config?

**Review by donpenney — CHANGES_REQUESTED** [review]

**Pull request merged**